framework/include/utils/MooseEnumBase.h

```diff
@@ -106,6 +106,15 @@ class MooseEnumBase
    */
   int getNextValidID() const;
 
+  /**
+   * Adds an enumeration item from name
+   */
+  MooseEnumBase & operator+=(const std::string & name);
+  /**
+   * Adds enumeration items from a list of names
+   */
+  MooseEnumBase & operator+=(const std::initializer_list<std::string> & names);
```

Comment:
initializer_list is mostly passed by value, as it is only a thin wrapper!

Reply:
Feel free to change it :-)